library/std/src/sys_common/rwlock.rs

```diff
@@ -26,6 +47,20 @@ impl RWLock {
         self.0.read()
     }
 
+    /// Acquires shared access to the underlying lock, blocking the current
+    /// thread to do so.
+    ///
+    /// The lock is automatically unlocked when the returned guard is dropped.
+    ///
+    /// Behavior is undefined if the rwlock has been moved between this and any
+    /// previous method call.
```

Comment:
If you make a separate `StaticRWLock` type that only has `&'static` methods, this method can be safe.